File: daily/2025/2529_Maximum_Count_of_Positive_Integer_and_Negative_Integer.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int maximumCount(vector<int>& nums) {
        int n = nums.size();
        if (nums[0] > 0 || nums[n-1] < 0) return n;
        int pos = n - (upper_bound(nums.begin(), nums.end(), 0) - nums.begin());
        int neg = lower_bound(nums.begin(), nums.end(), 0) - nums.begin();
        // cout << pos << " ," << neg << endl;
        return max(pos, neg);
    }
};

int main(){
    vector<int> nums = {-2,-1,-1,1,2,3};
    // vector<int> nums = {-3,-2,-1,0,0,1,2};
    // vector<int> nums = {5,20,66,1314};
    Solution S;
    cout << S.maximumCount(nums) << endl;
    return 0;
}

File: daily_python/2529_Maximum_Count_of_Positive_Integer_and_Negative_Integer.py
import os
import sys
from typing import List

class Solution:
    def lower_bound(self, nums, num):
        n = len(nums)
        left = 0
        right = n-1
        index = n
        while left <= right:
            mid = left + (right - left) // 2
            if nums[mid] >= num:
                index = mid
                right = mid - 1
            else:
                left = mid + 1
        return index
    
    def maximumCount(self, nums: List[int]) -> int:
        n = len(nums)
        if nums[0] > 0 or nums[n-1] < 0:
            return n
        pos = n - self.lower_bound(nums, 1)
        neg = self.lower_bound(nums, 0)
        return max(pos, neg)

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            dataset.append(nums)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums in dataset:
        results.append(solution.maximumCount(nums))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')